Modules/Project/Entities/Project.php

```diff
@@ -55,6 +55,44 @@ public function repositories()
         return $this->hasMany(ProjectRepository::class);
     }
 
+    public function resourceRequirement()
+    {
+        return $this->hasMany(ProjectResourceRequirement::class);
+    }
+
+    public function getResourceRequirementByDesignation($designationName)
+    {
+        return $this->resourceRequirement()->where('designation', $designationName)->first();
+    }
+
+    public function getDeployedCountForDesignation($designation)
+    {
+        return DB::table('project_team_members')
+                    ->where('project_id', '=', $this->id)
+                    ->where('designation', '=', $designation)
+                    ->count();
```

Comment:
You can use the relationship here
```suggestion
        return $this->getTeamMembers()->where('designation', '=', $designation)->count();
```